kotlintest-extensions/src/test/kotlin/com/sksamuel/kt/extensions/time/ConstantNowExtensionsTests.kt

```diff
@@ -0,0 +1,267 @@
+        withConstantNow(now) {
+          OffsetTime.now() shouldBeSameInstanceAs now
+        }
+      }
+    
+      it("Should reverse to default behavior after execution") {
+        withConstantNow(now) { }
+        delay(10)
+      
+        OffsetTime.now() shouldNotBeSameInstanceAs now
+      }
+    }
+    describe("The ConstantNow extension function (ThaiBuddhistDate)") {
+    
+      val now = ThaiBuddhistDate.now()
+    
+      it("Should replace the ThaiBuddhistDate.now() with my own dateTime") {
+        withConstantNow(now) {
+          ThaiBuddhistDate.now() shouldBeSameInstanceAs now
+        }
+      }
+    
+      it("Should reverse to default behavior after execution") {
+        withConstantNow(now) { }
+        delay(10)
+      
+        ThaiBuddhistDate.now() shouldNotBeSameInstanceAs now
+      }
+    }
+    describe("The ConstantNow extension function (Year)") {
+    
+      val now = Year.now()
+    
+      it("Should replace the Year.now() with my own dateTime") {
+        withConstantNow(now) {
+          Year.now() shouldBeSameInstanceAs now
+        }
+      }
+    
+      it("Should reverse to default behavior after execution") {
+        withConstantNow(now) { }
+        delay(10)
+      
+        Year.now() shouldNotBeSameInstanceAs now
+      }
+    }
+    describe("The ConstantNow extension function (YearMonth)") {
+    
+      val now = YearMonth.now()
+    
+      it("Should replace the YearMonth.now() with my own dateTime") {
+        withConstantNow(now) {
+          YearMonth.now() shouldBeSameInstanceAs now
+        }
+      }
+    
+      it("Should reverse to default behavior after execution") {
+        withConstantNow(now) { }
+        delay(10)
+      
+        YearMonth.now() shouldNotBeSameInstanceAs now
+      }
+    }
+    describe("The ConstantNow extension function (ZonedDateTime)") {
+    
+      val now = ZonedDateTime.now()
+    
+      it("Should replace the ZonedDateTime.now() with my own dateTime") {
+        withConstantNow(now) {
+          ZonedDateTime.now() shouldBeSameInstanceAs now
+        }
+      }
+    
+      it("Should reverse to default behavior after execution") {
+        withConstantNow(now) { }
+        delay(10)
+      
+        ZonedDateTime.now() shouldNotBeSameInstanceAs now
+      }
+    }
+  }
+}
+
+class ConstantNowExtensionsListenerTest : StringSpec() {
+
+  private val myNow = HijrahDate.now()
+  private val myNow2 = LocalDateTime.now()
+  
+  init {
+    "Should use my now" {
+      HijrahDate.now() shouldBeSameInstanceAs myNow
+      LocalDateTime.now() shouldBeSameInstanceAs myNow2
+    }
+  }
+  
+  override fun afterSpecClass(spec: Spec, results: Map<TestCase, TestResult>) {
+    HijrahDate.now() shouldNotBeSameInstanceAs myNow
+    LocalDateTime.now() shouldNotBeSameInstanceAs myNow2
+  }
+  
+  override fun listeners() = listOf(
+          ConstantNowTestListener(myNow),
+          ConstantNowTestListener(myNow2)
+  )
+}
```

Comment:
Really good idea and clever implementation. We can put this in 3.3 if you like.

Reply:
Sure!

My only issues with the implementation is that technically speaking, `Temporal` doesn't force any implementation to contain a static `now`. All `Temporal`s from Java do, but maybe a random temporal appears, and it won't work.

I said that in the docs, but idk if that's a concern.

I didn't want to create 17 implementations of 17 types with 17 ultra-similar executions